rust/tool/src/signature.rs

```diff
@@ -61,10 +59,8 @@ impl KeyPair {
             if std::io::stderr().write_all(&output.stderr).is_err() {
                 return false;
             };
-            println!(
-                "Failed to verify signature using sbverify with args `{:?}`",
-                &args
-            );
+            log::info!("Failed to verify signature of {path:?}.");
```

Comment:
This sounds like `error!` or at least `warn!`? I mean if this is not an error what is? :) In case of errors, it also makes sense to dump all information required for debugging that you have at hand immediately.

Reply:
But maybe the log message should be at the call site of `verify`, otherwise legitimate uses of verify might spam the log?

Reply:
Right now this function is called only to check if the systemd-boot binary is signed or not. There, it should not be `error` because we just update the binary if it not signed (e.g. because you just installed Lanzaboote). So you're right it should be at the call site. Maybe a warning makes more sense than an info. But there is really nothing for the user to do, that's why it maybe should be an info?